### ISSUE TITLE
why re-running the policy 

### ISSUE THREAD

**Issue by ImolaFodor** [nishantkr18/guided-cost-learning#2]
@nishantkr18 why not passing also the probs/logits through the generate_session function, instead of re-running the policy with the states in order to be able to compose the loss function on the policy?

**Comment by pgoudet-42** [comment]
@ImolaFodor  How would you do it? because the generate_session is called in a list anyway. \calling the policy in the list comprehension or n the for is almost the same. One thing i don't get in this implementation @nishantkr18  is why the mean cost per 1 game increases? shouldn't it decrease?
